skipruntime-ts/tests/src/tests.ts

```diff
@@ -759,6 +759,55 @@ const mapWithExceptionOnExternalService: SkipService<Input_SN, Input_SN> = {
   },
 };
 
+//// testSessions
+
+type Post = {
+  title: string;
+  body: string;
+};
+type PostWithFoo = Post & { foo: string | null };
+
+class SessionsMapper implements Mapper<number, Post, number, PostWithFoo> {
+  private readonly session: { foo: string } | null;
+
+  constructor(session: Json | null) {
+    this.session = session ? (session as { foo: string }) : null;
+  }
+
+  mapEntry(key: number, values: Values<Post>): Iterable<[number, PostWithFoo]> {
+    return values
+      .toArray()
+      .map((v: Post) => [key, { ...v, foo: this.session?.foo ?? null }]);
+  }
```

Comment:
Would something like this work?  i.e. checking instead of casting to the type of the session, and extracting `foo` in the constructor instead of every map.

```suggestion
  private readonly sessionData: string | null;

  constructor(session: Json | null) {
    this.session = session?.foo ?? null;
  }

  mapEntry(key: number, values: Values<Post>): Iterable<[number, PostWithFoo]> {
    return values
      .toArray()
      .map((v: Post) => [key, { ...v, foo: this.sessionData }]);
  }
```

Reply:
Yes, good call.